SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -217,10 +217,36 @@ bool USpatialGameInstance::ProcessConsoleExec(const TCHAR* Cmd, FOutputDevice& A
 	return false;
 }
 
+namespace
+{
+constexpr uint8 SimPlayerErrorExitCode = 10;
+
+void HandleOnSimulatedPlayerTravelFailure(UWorld* World, ETravelFailure::Type TravelType, const FString& Reason)
+{
+	UE_LOG(LogSpatialGameInstance, Log, TEXT("SimulatedPlayer failed to travel due to: %s"), *Reason);
+
+	FPlatformMisc::RequestExitWithStatus(/*bForce =*/false, SimPlayerErrorExitCode);
+}
+
+void HandleOnSimulatedPlayerNetworkFailure(UWorld* World, UNetDriver* NetDriver, ENetworkFailure::Type NetworkFailureType,
+										   const FString& Reason)
+{
+	UE_LOG(LogSpatialGameInstance, Log, TEXT("SimulatedPlayer network failure due to: %s"), *Reason);
+
+	FPlatformMisc::RequestExitWithStatus(/*bForce =*/false, SimPlayerErrorExitCode);
+}
+} // namespace
+
 void USpatialGameInstance::Init()
 {
 	Super::Init();
 
+	if (IsSimulatedPlayer())
```

Comment:
Maybe should replace this check with a command line flag check such as `-FailOnTravelError`, and pass it from the coordinator? This could make it easier to follow what happens, and would solve the issue of someone, say, launching these outside of the coordinator with hand-picked flags and wondering why the application shuts down on invalid deployment name, or in case the simplayer doesn't have the correct map packaged.

Reply:
The main thing is that this changes the expected behaviour as compared to UE4, and can't be opted-out. Maybe a flag in GameInstance (say, `bShouldFailOnTravelError`) would be an easier solution?

Reply:
I agree that passing a command line flag would be better.